sdk/src/main/java/com/mapbox/maps/MapSurface.kt

```diff
@@ -23,42 +23,62 @@ import com.mapbox.maps.renderer.widget.Widget
  *
  * <strong>Warning:</strong> Please note that you are responsible for getting permission to use the map data,
  * and for ensuring your use adheres to the relevant terms of use.
- *
- * @param context the application context to init the default [MapInitOptions]
- * @param surface the surface that will display map
- * @param mapInitOptions the init options to for map
  */
-class MapSurface @JvmOverloads constructor(
-  // could use strong ref here as MapInitOptions have strong ref in any case
-  private val context: Context,
-  val surface: Surface,
-  mapInitOptions: MapInitOptions = MapInitOptions(context)
-) : MapPluginProviderDelegate, MapControllable {
+class MapSurface : MapPluginProviderDelegate, MapControllable {
 
+  private val context: Context
   private val mapController: MapController
-  private val renderer: MapboxSurfaceRenderer = MapboxSurfaceRenderer(mapInitOptions.antialiasingSampleCount)
-
-  init {
-    this.mapController = MapController(
-      renderer,
-      mapInitOptions
-    )
-    mapController.initializePlugins(mapInitOptions)
-  }
-
-  /**
-   * Called when the surface has been created.
+  private val mapInitOptions: MapInitOptions
+  private val renderer: MapboxSurfaceRenderer
+  val surface: Surface
+
+  /**
+   * @param context the application context to init the default [MapInitOptions]
+   * @param surface the surface that will display map
+   * @param mapInitOptions the init options for map
+   */
+  @JvmOverloads constructor(
+    context: Context,
+    surface: Surface,
+    mapInitOptions: MapInitOptions = MapInitOptions(context) // could use strong ref here as MapInitOptions have strong ref in any case
+  ) {
+    this.context = context
+    this.surface = surface
+    this.mapInitOptions = mapInitOptions
+    this.renderer = MapboxSurfaceRenderer(mapInitOptions.antialiasingSampleCount)
+    this.mapController = MapController(renderer, mapInitOptions).apply {
+      initializePlugins(mapInitOptions)
+    }
+  }
+
+  @VisibleForTesting(otherwise = PRIVATE)
+  internal constructor(
+    context: Context,
+    surface: Surface,
+    mapInitOptions: MapInitOptions,
+    renderer: MapboxSurfaceRenderer,
+    mapController: MapController
+  ) {
+    this.context = context
+    this.surface = surface
+    this.mapInitOptions = mapInitOptions
+    this.renderer = renderer
+    this.mapController = mapController
+  }
+
+  /**
+   * Must be called when the surface has been created.
    */
   fun surfaceCreated() {
     renderer.surfaceCreated()
     // display should not be null at this point but to be sure we will fallback to DEFAULT_FPS
-    val screenRefreshRate = (context.getSystemService(Context.WINDOW_SERVICE) as WindowManager)
-      .defaultDisplay?.refreshRate?.toInt() ?: MapView.DEFAULT_FPS
+    val screenRefreshRate = (context.getSystemService(Context.WINDOW_SERVICE) as WindowManager?)
```

Comment:
is `WindowManager?` required here?

Reply:
Mocked `Context` does not return any real WindowManager, this does not harm to have in the code base IMO.

Reply:
but why not to mock context method?
```
val manager = mockk<WindowManager>()
...
every { context.getSystemService(Context.WINDOW_SERVICE) } returns manager
```

Reply:
In theory, any requested object from getSystemService _could_ be null, so maybe it is safer to allow for it in the SDK?

Reply:
Definitely can. But also nice to check that we're calling the correct method.

Reply:
I'll use your suggestion but it might be nice to keep the null-check in the sdk itself.

Reply:
yeah, good idea to check null-values from the system